Lib/typing.py

```diff
@@ -683,6 +683,9 @@ def copy_with(self, params):
         return _GenericAlias(self.__origin__, params, name=self._name, inst=self._inst)
 
     def __repr__(self):
+        if (self.__origin__ == Union and len(self.__args__) == 2
+                and self.__args__[1] is type(None)):
```

Comment:
Hi @Endilll, thanks for opening a pull request for this. Notice that this does not work, when `None` is the first element:
```
>>> import typing
>>> typing.Union[None, int]
typing.Union[NoneType, int]
>>> typing.Union[None, int] == typing.Optional[int]
True
```

This change will also need some tests and an NEWS entry.

Reply:
Thank you for the feedback. Fixed